### DISCUSSION TITLE
Welcome to Django-CryptographicFields Discussions!

### DISCUSSION THREAD

**Discussion by shah-dhwanil** [Shahprogrammer/Django-CryptographicFields#12] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by micha-s** [comment]
May I ask if there is a reason why there are no 'exact'/'iexact' filters?

**Reply by shah-dhwanil** [reply]
Because it is indirectly done by django 's filter method.

**Reply by micha-s** [reply]
I have encrypted the email field of the django user:
```python
class User(AbstractUser):
    email = encrypt.EmailField(_('email address'), blank=True)
    ...
```
Now in `django.contrib.auth.forms.PasswordResetForm.get_users` the `filter(email__iexact=email)` filter is used. And it fails to get any matches. I tried it and it appears that `filter(email__istartswith=email)` also doesn't work. But `filter(email=email)` works! (In this test I made sure that the character cases are matching.)

But now the weirdest part: If I am stepping through with a debugger all filters above are working as intended. 

Has anyone information to solve this puzzle?

**Reply by shah-dhwanil** [reply]
You can't use filters which are provided with Django. You need to use filters provided in this package

**Reply by shah-dhwanil** [reply]
You can't use istartwith directly because for Capital string the encrypted message will be different from the encrypted message of lower case.
If you want to use istartswith you need to use ``` 
CryptographicFields.filters.istartswith```.

**Reply by shah-dhwanil** [reply]
To use this filters refer https://django-cryptographicfields.readthedocs.io/en/latest/examples.html#sorting-by-example

**Reply by micha-s** [reply]
Thank you! I missed that information. I thought the proper filter gets selected automatically.
Now I can write a workaround so that `django.contrib.auth.forms.PasswordResetForm.get_users` works again.
But is it possible to make the filters getting selected automatically? Maybe like you did in `fields.py` with `EmailField.register_lookup(StartsWith)`? Or maybe by overriding some methods of the inherited `django.db.models.query_utils.RegisterLookupMixin` in the encrypted field classes?
My intent is to get all django code working without subclassing (e.g. `PasswordResetForm`).

**Reply by shah-dhwanil** [reply]
Thanks for idea. I will work on it if it is possible
